op-service/dial/l2_provider.go

```diff
@@ -0,0 +1,69 @@
+package dial
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type L2EndpointProvider interface {
+	EthClient(ctx context.Context) (*ethclient.Client, error)        // op-geth
+	RollupClient(ctx context.Context) (*sources.RollupClient, error) // op-node
```

Comment:
This works now, as there aren't any tests for the `StaticL2EndpointProvider`, but when you get to implementing the active provider, we might need to change this interface to return interfaces of the eth and rollup client for testability (so you can mock `SequencerActive` calls etc in tests). It's not nice to return interfaces, but not sure how to achieve good testability while also returning struct types.

So no action for this PR, but wanted to flag this.